public/src/components/switchboard.tsx

```diff
@@ -0,0 +1,147 @@
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Switch from '@material-ui/core/Switch';
+import Button from '@material-ui/core/Button';
+import SaveIcon from '@material-ui/icons/Save';
+
+
+interface Switches {
+  [key: string]: boolean
+  stripeOneOff: boolean,
+  payPalOneOff: boolean,
+  stripeRecurring: boolean,
+  payPalRecurring: boolean,
+  directDebitRecurring: boolean
+}
+
+function safelyExtractSwitchValue(json: object, path: string): boolean {
+  const switchValue = get(json, path, false);
+  return switchValue === 'On';
+}
+
+function jsonToSwitches(json: object): Switches {
+  return {
+    stripeOneOff: safelyExtractSwitchValue(json, 'value.switches.oneOffPaymentMethods.stripe'),
+    payPalOneOff: safelyExtractSwitchValue(json, 'value.switches.oneOffPaymentMethods.payPal'),
+    stripeRecurring: safelyExtractSwitchValue(json, 'value.switches.recurringPaymentMethods.stripe'),
+    payPalRecurring: safelyExtractSwitchValue(json, 'value.switches.recurringPaymentMethods.payPal'),
+    directDebitRecurring: safelyExtractSwitchValue(json, 'value.switches.recurringPaymentMethods.directDebit')
+  };
+}
+
+export class Switchboard extends React.Component {
+  state: Switches;
+  previousStateFromServer: object;
+
+  constructor(props: {}) {
+    super(props);
+    this.state = {
+      stripeOneOff: false,
+      payPalOneOff: false,
+      stripeRecurring: false,
+      payPalRecurring: false,
+      directDebitRecurring: false
+    };
+    this.previousStateFromServer = {};
+  }
+  
+  componentWillMount(): void {
+    this.fetchStateFromServer();
+  }
+
+  fetchStateFromServer(): void {
+    fetch('/support-frontend/switches')
+      .then(resp => resp.json())
+      .then(json => {
+        this.previousStateFromServer = json;
+        return json;
+      })
+      .then(jsonToSwitches)
+      .then(switches => {
+        this.setState(switches);
+      });
+  }
+
+  updateSwitch = (switchName: string) => (event: React.FormEvent<HTMLInputElement>) => {
+    // it's annoying I have to do this type cast here
+    this.setState({ [switchName]: (event.target as HTMLInputElement).checked });
+  };
+
+  saveSwitches = () => {
+    const newState = {...this.previousStateFromServer};
+
+    // actually I'm not sure I like this technique since it bypasses the type system
+    // though in this case it doesn't matter since it's just an object type
+    set(newState, 'value.switches.oneOffPaymentMethods.stripe', this.state.stripeOneOff ? 'On' : 'Off');
+    set(newState, 'value.switches.oneOffPaymentMethods.payPal', this.state.payPalOneOff ? 'On' : 'Off');
+    set(newState, 'value.switches.recurringPaymentMethods.stripe', this.state.stripeRecurring ? 'On' : 'Off');
+    set(newState, 'value.switches.recurringPaymentMethods.payPal', this.state.payPalRecurring ? 'On' : 'Off');
+    set(newState, 'value.switches.recurringPaymentMethods.directDebit', this.state.directDebitRecurring ? 'On' : 'Off');
```

Comment:
is this form going to allow us to set other switches? Currently the file also has `optimize` and `experiments`
If this component is going to update all settings in this file then it might be worth defining the type of this object properly

Reply:
I may now disagree with my statement from yesterday - if it means a load of boilerplate code then the server-side validation is enough!